pool/test/authorization_lists_test.ml

```diff
@@ -0,0 +1,282 @@
+  let%lwt exp2 = ExperimentRepo.create () in
+  let%lwt session2 = SessionRepo.create ~location exp2 () in
+  let%lwt (_ : Assignment.t) = AssignmentRepo.create session2 contact2 in
+  let%lwt () = assign_role actor `LocationManager (Some (location_target location)) in
+  let%lwt contacts = get_by_user () in
+  let expected = [ contact1; contact2 ] |> sort in
+  check testable "Contacts 1, 2 returned" expected contacts;
+  (* With actor permission *)
+  let%lwt contact3 = ContactRepo.create () in
+  let%lwt () =
+    create_actor_permission actor Guard.Permission.Read (contact_target contact3)
+  in
+  let%lwt contacts = get_by_user () in
+  let expected = [ contact1; contact2; contact3 ] |> sort in
+  check testable "Contacts 1, 2, 3 returned" expected contacts;
+  (* As recruiter *)
+  let%lwt () = assign_role actor `Recruiter None in
+  let query =
+    let open Query in
+    create ~pagination:(Pagination.create ~item_count:1 ()) ()
+  in
+  let%lwt expected = Contact.all ~query pool ||> query_item_count in
+  let%lwt contact_count = Contact.list_by_user ~query pool actor ||> query_item_count in
+  check int "All contacts are returned" expected contact_count;
+  Lwt.return_unit
+;;
+
+let admins _ () =
+  let testable = list Test_utils.admin in
+  let sort = CCList.sort Admin.compare in
+  let%lwt actor = create_actor () in
+  let get_by_user () = Admin.list_by_user pool actor ||> fst ||> sort in
+  (* Without any roles *)
+  let%lwt admins = get_by_user () in
+  check testable "No admins returned" [] admins;
+  (* As assistant *)
+  let%lwt () = assign_role actor `Assistant None in
+  let%lwt admins = get_by_user () in
+  check testable "No admins returned" [] admins;
+  (* With actor permission *)
+  let%lwt admin = Integration_utils.AdminRepo.create () in
+  let%lwt () = create_actor_permission actor Guard.Permission.Read (admin_target admin) in
+  let%lwt admins = get_by_user () in
+  check testable "One admin returned" [ admin ] admins;
+  (* As recruiter *)
+  let%lwt () = assign_role actor `Recruiter None in
+  let query =
+    let open Query in
+    create ~pagination:(Pagination.create ~item_count:1 ()) ()
+  in
+  let%lwt expected = Admin.all ~query pool ||> query_item_count in
+  let%lwt admin_count = Admin.list_by_user ~query pool actor ||> query_item_count in
+  check int "All admins are returned" expected admin_count;
+  Lwt.return_unit
+;;
+
+module CalendarUtils = struct
+  open Session
+
+  let testable = list Id.(testable pp equal)
+  let sort = CCList.sort Id.compare
+  let session_ids = CCList.map (fun { id; _ } -> id)
+  let calendar_session_ids = CCList.map (fun (t : Calendar.t) -> t.Calendar.id)
+  let start_time = Ptime_clock.now ()
+
+  let end_time =
+    Ptime.add_span start_time Sihl.Time.(duration_to_span OneWeek)
+    |> CCOption.get_exn_or "Invalid timespan"
+  ;;
+
+  let start = Test_utils.Model.in_an_hour ()
+end
+
+let dashboard_calendar _ () =
+  (* TODO: Compare generated links *)
```

Comment:
Done in a followup PR: https://github.com/uzh/z-pool-tool/blob/bug/2359-session-links/pool/test/authorization_lists_test.ml